src/rng-browser.js

```diff
@@ -4,18 +4,22 @@
 
 // getRandomValues needs to be invoked in a context where "this" is a Crypto implementation. Also,
 // find the complete implementation of crypto (msCrypto) on IE11.
-var getRandomValues =
-  (typeof crypto != 'undefined' && crypto.getRandomValues && crypto.getRandomValues.bind(crypto)) ||
-  (typeof msCrypto != 'undefined' &&
-    typeof msCrypto.getRandomValues == 'function' &&
+const getRandomValues =
+  (typeof crypto !== 'undefined' &&
+    crypto.getRandomValues &&
+    crypto.getRandomValues.bind(crypto)) ||
+  (typeof msCrypto !== 'undefined' &&
+    typeof msCrypto.getRandomValues === 'function' &&
     msCrypto.getRandomValues.bind(msCrypto));
 
-var rnds8 = new Uint8Array(16); // eslint-disable-line no-undef
 export default function rng() {
   if (!getRandomValues) {
     throw new Error(
       'crypto.getRandomValues() not supported. See https://github.com/uuidjs/uuid#getrandomvalues-not-supported',
     );
   }
+
+  const rnds8 = new Uint8Array(16);
```

Comment:
This might have a performance impact. A new `Uint8Array` is created with each call to `rng()` instead of reusing a preallocated array. Please verify!

Reply:
I think this is unsafe, because someone can use your `rng()` function in their asynchronous code. When generating random numbers, it will always receive the same array. (New data will overwrite old).

Multiple function calls will return the same array.

Reply:
`rng-browser` is not part of the public API, so should not be used externally.  Internally, there's no async code that uses it.  It's also very purpose-specific (generating a 16-byte array of random #'s), so I don't see there being a lot of use cases where someone would actually need to figure out how to reach into this module to access this function.  Thus, the risk to users feels more hypothetical than real.

That said, reusing the array was an optimization I put in place [a long time ago](https://github.com/uuidjs/uuid/commit/774c6e3169dd04be2a80af14153d9379586dd064#diff-bef26ab2ce6b354722523f65bc301d59R16).  There was no real demand for it (nobody was complaining about perf at the time).  It was just a vanity thing to see how much perf I could squeeze out of the code.

Now, _having said that_, it's looks like it's still a hell of an optimization, which I'll suggest is worth keeping.

[/me pats himself on back 😆]

On `master`:
```
rng() reusing array:
uuidv4() x 507,182 ops/sec ±5.43% (63 runs sampled)
benchmark.js:32 uuidv4() fill existing array x 688,903 ops/sec ±8.40% (63 runs sampled)

rng() creating new array each time:
uuidv4() x 183,198 ops/sec ±10.90% (45 runs sampled)
benchmark.js:32 uuidv4() fill existing array x 211,791 ops/sec ±13.46% (41 runs sampled)
```

On this PR:
```
rng() reusing array:
uuidv4() x 524,604 ops/sec ±0.82% (64 runs sampled)
benchmark.js:20 uuidv4() fill existing array x 741,362 ops/sec ±0.52% (64 runs sampled)

rng() creating new array each time:
uuidv4() x 197,603 ops/sec ±10.09% (51 runs sampled)
benchmark.js:20 uuidv4() fill existing array x 211,450 ops/sec ±15.85% (41 runs sampled)
```

(Note: I suspect the larger margin of error in the new-array case is due is because the increased level of array creation results in more erratic garbage collection.)

Reply:
> [/me pats himself on back 😆]

Well played, 9-year-ago-you, well played! 😂

Reply:
@awwit please re-introduce this optimization by moving the array definition out of the function scope again.